packages/react/src/editor/BlockNoteDefaultUI.tsx

```diff
@@ -29,7 +70,16 @@ export function BlockNoteDefaultUI(props: BlockNoteDefaultUIProps) {
       {props.formattingToolbar !== false && <FormattingToolbarController />}
       {props.linkToolbar !== false && <LinkToolbarController />}
       {props.slashMenu !== false && (
+        <>
         <SuggestionMenuController triggerCharacter="/" />
+        {/*STEP 1:  the suggestion menu for the emojis, with custom component and getItems */}
+        <SuggestionMenuController
```

Comment:
Couple of things:
- I think we should get rid of the `isEmoji` property and just use an `onItemClick` handler here instead of having the logic to insert emojis in the `SuggestionMenuController`
- let's put this behind a flag (`props.emojiMenu` instead of `props.slashMenu`)
- bit of cleanup, let's separate the Emoji related code to a `<EmojiMenuController>` (init, search, etc) so that we don't have so much emoji related code in `BlockNoteDefaultUI`?

Reply:
the thing is, despite the simulation of colon key, the rest of the stuff is still working as a custom suggestion menu. The only reason I added this simulation was to make sure the emoji menu was accessible thru the slash menu, otherwise it was working perfect as a custom suggestion menu handled by the `suggestionMenuController`. Then when I moved towards the slash menu thing, I first tried rendering the colon `inline` to trigger the emoji menu, which didn't work, leaving me with the option of simulation. But still it couldn't trigger the `handleTextInput` as it only responds when the user directly inputs (see prosemirror docs). Therefore I came up with a hack to first trigger the `onKeyPress`, then call the rest of the logic which is responsible for showing the menu. Both things have to exist in order to make it work. It will require collaboration and some time to take a different approach, may even require a separate issue.

Reply:
@matthewlipski could you help with a programmatic way to open the menu?

Reply:
Sure! If you look in the side menu, notice we can press the `+` button to open the slash menu without having to press the `/` key. This is done with the following code:
```tsx
this.pmView.dispatch(
  this.pmView.state.tr.scrollIntoView().setMeta(suggestionMenuPluginKey, {
    triggerCharacter: "/",
    fromUserInput: false,
  })
);
```
All you should need to do to replicate this for the emoji menu is change `triggerCharacter: "/"` to `triggerCharacter: ":"`. Then we can get rid of all the extra code for simulating the colon input.